### DISCUSSION TITLE
The loss of on_stage_start

### DISCUSSION THREAD

**Discussion by MaizeKai** [speechbrain/speechbrain#1798] [Q&A] [answered]
Hello,I would like to inquire about the loss incurred during the on_stage_start phase, "The first image depicts the enhancement model that utilizes the VoiceBank dataset, The loss can be fined in the stage.Train between lines 95 and 106.

![image](https://user-images.githubusercontent.com/109297054/211766356-948f1810-6581-4079-950b-7df7e2eaeac7.png)

The keyword spotting model which is trained on the Google Speech Command v0.02 dataset, no loss function was implemented in the stage.Train.

![image](https://user-images.githubusercontent.com/109297054/211768414-86f8313b-1671-466a-82f5-39b7459888e7.png)

Can you explain the distinctions between the two models?

Any assistance would be greatly appreciated!

**Comment by TParcollet** [comment, marked as answer]
Hi, SpeechBrain does not enforce any ontology about where things should go. This is our opinion, the major strength of the toolkit compared to others: you can basically do anything.

The most common behavior w.r.t to losses is to see it in the compute_objective function. The second most common is to have it in the compute_forward. But if one wants to "instantiate" the loss, because say, one has a loss with some persistent states that must be initialised once, then it is appropriate to put it in a start of epoch callback (although I don't know about pesq).

@pplantinga might be able to answer precisely on that point.

**Reply by MaizeKai** [reply]
I got it, thank you for your answer!

**Comment by pplantinga** [comment]
Neither of the snippets you provided do any actual loss computation. These are purely set-up functions for metric (not loss) computation. Different metrics are useful for different tasks, and computing metrics on train set is completely optional (unlike loss).

**Reply by MaizeKai** [reply]
Thanks for your answer!!
